### PR TITLE
docs: use American English instead of British English

### PR DESCRIPTION
**Pull request by dundargoc**
This makes docs more consistent and more greppable.

This is primarily focused on words ending on "our" in British English.

### CODE REVIEW

**Comment by clason** [comment]
Maybe add to the [cringelist](https://github.com/neovim/neovim/blob/86e54734bf9e05605d8b7146e7b0e79025138ba2/scripts/gen_help_html.lua#L37-L47)?

**Comment by dundargoc** [comment]
Yeah, we might need a language police CI to ban british if we want to prevent regressions but maybe that's too much lol

**Review by bfredl — APPROVED** [review]
image

**Comment by clason** [comment]
Do `neighbour` next?

**Comment by justinmk** [comment]
biscuits => cookies, chips => fries, crumpets => english muffins

**Comment by clason** [comment]
you forgot trousers => pants 😤

File: scripts/gen_help_html.lua [hunk 1]
```diff
@@ -44,6 +44,14 @@ local spell_dict = {
   viml = 'Vimscript',
   ['tree-sitter'] = 'treesitter',
   ['Tree-sitter'] = 'Treesitter',
+  colour = 'color',
```
Comment (dundargoc) [inline scripts/gen_help_html.lua:47]: TODO: Add a title-case invariant check/regular word dict to avoid duplicating every word

**Comment by craigmac** [comment]
makes sense,`:colourscheme` and some others were never a thing!

**Comment by dundargoc** [comment]
I can't come up with a non-biased way to say "using 'ou' is super ugly and morally wrong" so I guess I'll just leave that out of the commit message.

**Comment by zeertzjq** [comment]
These changes will make merging documentation in Vim patches harder

**Comment by zeertzjq** [comment]
And... Why are you changing the oldtests as well?

**Comment by clason** [comment]
...unless someone were to make the same changes in Vim. Presumably, they care about consistent documentation as well.

**Comment by dundargoc** [comment]
I don't think this is a change vim would be interested in.

**Pull request closed without merging**